OpenRA.Mods.Common/Warheads/CreateResourceWarhead.cs

```diff
@@ -17,43 +17,64 @@
 
 namespace OpenRA.Mods.Common.Warheads
 {
-	public class CreateResourceWarhead : Warhead
+	public class CreateResourceWarhead : Warhead, IRulesetLoaded<WeaponInfo>
 	{
 		[Desc("Size of the area. The resources are seeded within this area.", "Provide 2 values for a ring effect (outer/inner).")]
 		public readonly int[] Size = { 0, 0 };
 
 		[Desc("Will this splatter resources and which?")]
 		public readonly string AddsResourceType = null;
 
+		void IRulesetLoaded<WeaponInfo>.RulesetLoaded(Ruleset rules, WeaponInfo info)
+		{
+			if (string.IsNullOrEmpty(AddsResourceType))
```

Comment:
Wouldn't it make more sense to `[FieldLoader.Require]` the field and remove this check? I don't see a use case for having a warhead that doesn't create anything.

Reply:
I would have done so, if that didn't conflict with #18012 (we'd have to remove `[FieldLoader.Require]` again there).

Reply:
IMO it would make sense to require the resource types to be defined for removal too